---
layout: default
title: Publications
modified: 2/19/2026
description: Papers, talks, and posters about GRNsight, a web application and service for visualizing models of small-scale gene regulatory and protein-protein interaction networks.
---
            <p id="pageTitle">Publications</p>

            <div id="pageContent">
                <div class="panel-group">
                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <p class="panel-title aboutHeaders">
                                <a href="#papers" data-toggle="collapse">
                                    Papers
                                </a>
                            </p>
                        </div>

                        <div id="papers" class="panel-collapse collapse in">
                            <div class="panel-body">
                                
                                <p class="publication">
                                    <b>September 12, 2016</b><br>
                                    Kam D. Dahlquist, John David N. Dionisio, Ben G. Fitzpatrick, Nicole A. Anguiano, Anindita Varshneya, Britain J. Southwick, Mihir Samdarshi. (2016) GRNsight: a web application and service for visualizing models of small- to medium-scale gene regulatory networks. <em>PeerJ Computer Science</em> 2:e85 <a href="https://doi.org/10.7717/peerj-cs.85" target="_blank">https://doi.org/10.7717/peerj-cs.85</a><br><br>

                                    <a href="https://doi.org/10.7717/peerj-cs.85" target="_blank">View HTML with embedded PDF</a><br>
                                    <a href="https://peerj.com/articles/cs-85.pdf" target="_blank">Download PDF (2.7 MB)</a><br>
                                    <a href="https://peerj.com/articles/cs-85/reviews/" target="_blank">See peer review history</a><br><br>
                                    
                                    <i>Note that preprints of this article were posted in PeerJ Preprints on <a href="https://peerj.com/preprints/2068v1/" target="_blank">May 23</a> and <a href="https://peerj.com/preprints/2068v2/" target="_blank">August 18</a>, 2016.</i><br><br>

                                    <b>Abstract</b><br>
                                    GRNsight is a web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. The original motivation came from our efforts to perform parameter estimation and forward simulation of the dynamics of a differential equations model of a small GRN with 21 nodes and 31 edges. We wanted a quick and easy way to visualize the weight parameters from the model which represent the direction and magnitude of the influence of a transcription factor on its target gene, so we created GRNsight. GRNsight automatically lays out either an unweighted or weighted network graph based on an Excel spreadsheet containing an adjacency matrix where regulators are named in the columns and target genes in the rows, a Simple Interaction Format (SIF) text file, or a GraphML XML file. When a user uploads an input file specifying an unweighted network, GRNsight automatically lays out the graph using black lines and pointed arrowheads.  For a weighted network, GRNsight uses pointed and blunt arrowheads, and colors the edges and adjusts their thicknesses based on the sign (positive for activation or negative for repression) and magnitude of the weight parameter. GRNsight is written in JavaScript, with diagrams facilitated by D3.js, a data visualization library. Node.js and the Express framework handle server-side functions. GRNsight’s diagrams are based on D3.js’s force graph layout algorithm, which was then extensively customized to support the specific needs of GRNs. Nodes are rectangular and support gene labels of up to 12 characters.  The edges are arcs, which become straight lines when the nodes are close together.  Self-regulatory edges are indicated by a loop.  When a user mouses over an edge, the numerical value of the weight parameter is displayed. Visualizations can be modified by sliders that adjust the force graph layout parameters and through manual node dragging. GRNsight is best-suited for visualizing networks of fewer than 35 nodes and 70 edges, although it accepts networks of up to 75 nodes or 150 edges. GRNsight has general applicability for displaying any small, unweighted or weighted network with directed edges for systems biology or other application domains. GRNsight serves as an example of following and teaching best practices for scientific computing and complying with FAIR Principles, using an open and test-driven development model with rigorous documentation of requirements and issues on GitHub. An exhaustive unit testing framework using Mocha and the Chai assertion library consists of around 160 automated unit tests that examine nearly 530 test files to ensure that the program is running as expected. The GRNsight application (<a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>) and code (<a href="https://github.com/dondi/GRNsight">https://github.com/dondi/GRNsight</a>) are available under the open source BSD license.
                                </p>

                            </div>
                        </div>
                    </div>

                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <p class="panel-title aboutHeaders">
                                <a href="#presentations" data-toggle="collapse">
                                    Presentations
                                </a>
                            </p>
                        </div>

                        <div id="presentations" class="panel-collapse collapse">
                            <div class="panel-body">

                                <p class="abstract">
                                    <b><a href="https://www.sccur.org/">Southern California Conference for Undergraduate Research (SCCUR) 2025, California State University Channel Islands, Camarillo, CA</a></b><br><br>
                                    November 22, 2025<br>
                                    GRNsight v7: a web application for visualizing gene regulatory networks and protein-protein interactions<br>
                                    <b>Ngoc K. Tran, Milka Zekarias</b>, Cecilia J. Zaragoza, A'Kaia Phelps, Amelie Dinh, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is an open-source web application for visualizing models of gene regulatory networks (GRNs) and protein-protein physical interaction networks (PPIs) either uploaded by the user or obtained from our backend yeast database. Models are displayed as graphs where genes/proteins are rectangular nodes with edges between them. GRNs have directed edges to indicate regulatory relationships (activation or repression), and PPIs have undirected edges to indicate physical binding. For v7, we improved our PPI support, transitioned our database loading scripts from YeastMine to AllianceMine, enhanced viewport functionality, and migrated continuous integration (CI) to GitHub Actions. As was previously implemented for GRNs, users can now color nodes in PPIs with expression data, either provided by the user or from our backend database. Different datasets can be displayed on the top and bottom of the nodes. For database loading, our scripts were ported to AllianceMine because YeastMine was discontinued. New graph functionality has been implemented that allows users to restrict the graph to viewport so that they can drag, zoom, and move the graph with the nodes staying within the application’s bounding box. Finally, the project’s CI pipeline has been migrated from Travis CI to GitHub Actions, integrating workflows for automated builds, unit tests, and linting to streamline development processes and improve maintainability. Ongoing efforts prioritize bug resolution, user interface enhancements, and improved documentation. GRNsight is now positioned to comprehensively address GRNs and PPIs, offering a unified platform for visualizing diverse molecular interaction models. GRNsight is available at <a href="https://dondi.github.io/GRNsight/">https://dondi.github.io/GRNsight/</a>. <br>
                                
                                     Click <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/presentations/Tran_Zekarias_SCCUR_20251122.pdf" target="_blank">here</a> to view the slides.      
                                </p>

                                <p class="abstract">
                                    <b>Pomona College Department of Biology, Claremont, California</b><br><br>
                                    September 26, 2025<br>
                                    From the <i>lac</i> operon to the spaghettiome, what can we learn from modeling gene regulatory networks on a medium scale?<br>
                                    <b>Kam D. Dahlquist</b><br><br>
                                </p>

                                <p class="abstract">
                                    <b>Southern California Systems Biology Conference, UC Irvine</b><br><br>
                                    March 2024<br>
                                    Dynamical systems modeling and visualization of yeast cold shock gene regulatory networks<br>
                                    <b>Kam D. Dahlquist</b><br><br>
                                </p>
                                        
                                <p class="abstract">
                                    <b>Joint Bioinformatics Community Conference and Bioinformatics Open Source Conference (virtual)</b><br><br>
                                    July 2020<br>
                                    New features for GRNsight:  a web application for visualizing models of small- to medium-scale gene regulatory networks<br>
                                    <b>Kam D. Dahlquist</b><br><br>
                                </p>

                                <p class="abstract">
                                    <b>12<sup>th</sup> Annual Loyola Marymount University Undergraduate Research Conference (virtual)</b><br><br>
                                    April 2020<br>
                                    An Expression Database for Node Coloring and Expansion of Species for Gene Pages for GRNsight<br>
                                    <b>Alexia M. Filler, Kevin B. Patterson</b>, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN, can mathematically model GRNs. Using a GRNmap output workbook, GRNsight automatically visualizes the network as a graph. GRNsight uses the model output to color-code the edges based on the activation and repression relationships between the transcription factors and colors the nodes based on time course gene expression data. We have added a feature to store expression data in a database, so that users of GRNsight can view and interact with already-existing expression data pulled from the Dahlquist lab and other publicly available datasets. The database is hosted by Amazon Web Services (AWS) Relational Database Service (RDS) and uses a PostgreSQL engine, from which GRNsight performs dynamic queries in response to user requests. Additionally, we have expanded the availability of gene-specific data to a greater number of species. The feature allows user to specify the species of interest either based on a GRNmap workbook or direct user input, and save it for the remainder of the user's session. With the species indicated, the gene page can query the appropriate databases and return information relevant only to species-specific genes. GRNsight is available at <a href="https://dondi.github.io/GRNsight/">https://dondi.github.io/GRNsight/</a><br>

                                            Click <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/presentations/Filler_Patterson_LMU-Symposium_20200418_talk.pdf" target="_blank">here</a> to view the slides. 
                                </p>

                                 <p class="abstract">
                                    <b>11<sup>th</sup> Annual Loyola Marymount University Undergraduate Research Conference</b><br><br>
                                    March 2019<br>
                                    New Layouts, Data Types, and Architecture for GRNsight 3:  a Web Application for Visualizing Gene Regulatory Networks<br>
                                    <b>Mihir Samdarshi</b>, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing small- to medium-scale gene regulatory networks (GRNs). A GRN consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNsight accepts input files in the Excel, SIF, and GraphML formats and then produces weighted or unweighted network graphs by representing genes as nodes and regulatory connections as edges. Activation is represented with red edges, repression is represented with blue edges, and weight magnitudes below a certain threshold are represented with gray edges to denote weak activation or repression relationships. GRNsight now contains features to facilitate network comparison. Nodes on each graph show a heat map corresponding to gene expression levels over time; these nodes can be overlaid with up to two different data sets. The graphs generated by GRNsight may now be automatically laid out in a grid. When a user right-clicks on a gene, data requests are made to five public databases and a web page is dynamically populated with the data returned. Finally, a number of under the hood improvements were made to GRNsight’s architecture. Most notably, GRNsight was made to follow a strict Model-View-Controller (MVC) paradigm. MVC consolidates the application’s state into a single object, from which presentation and interaction are derived. GRNsight version 3.1.3 is freely available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>; the code is available under the open source BSD license at <a href="https://github.com/dondi/GRNsight">https://github.com/dondi/GRNsight</a>.<br>

                                            Click <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/presentations/Samdarshi_LMU-Symposium_20190323_talk.pdf" target="_blank">here</a> to view the slides. 
                                </p>

                                <p class="abstract">
                                    <b>10<sup>th</sup> Annual Loyola Marymount University Undergraduate Research Conference</b><br><br>
                                    March 2018<br><br>
                                    Extending the Visualization Capabilities of GRNsight: a Web Application for Visualizing Models of Gene Regulatory Networks<br>
                                    <b>Eileen J. Choe</b>, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing medium-scale gene regulatory networks (GRNs). A GRN consists of genes, transcription factors, and the regulatory connections between them which control expression of mRNA and protein from genes. GRNsight produces weighted or unweighted network graphs by representing genes as nodes and regulatory connections as edges, with colors, end markers, and thicknesses corresponding to the sign and magnitude of activation or repression. GRNsight is designed to accept Excel workbooks from GRNmap, a MATLAB GRN modeling program, and lay out the network graph specified by an adjacency matrix contained within one of the worksheets. However, the GRNmap output contains additional worksheets which store data about the GRN and results of the GRNmap simulations that have not been displayed or visualized by GRNsight. To address this, we have redesigned the Excel spreadsheet importer to parse and store data from the entire GRNmap output workbook in a standardized JavaScript Object Notation (JSON) data structure. The complete data extraction from the GRNsight output includes experimental and simulated temporal expression data for each gene in the network. We have developed a coloring framework to visualize this expression data in the form of a stacked heat map overlay of the nodes which compares the experimental and simulated gene expression levels. Node coloring allows GRNsight users to better derive insights from the data that model these networks. The GRNsight application http://dondi.github.io/GRNsight/ and code https://github.com/dondi/GRNsight are available under the open source BSD license.<br><br>

                                    New Graph Layouts for GRNsight: a Web Application for Visualizing Models of Gene Regulatory Networks <br>
                                    <b>Yeon-Soo Shin</b>, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing small- to medium-scale gene regulatory networks (GRNs). A GRN consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNsight imports data from an adjacency matrix in either an Excel workbook, SIF, or GraphML file, then produces weighted or unweighted network graphs by representing genes as nodes and regulatory connections as edges, with colors, end markers, and thicknesses corresponding to the sign and magnitude of activation or repression. Visual issues regarding edge thickness and end markers have been adjusted and test files created. The new GRNsight graph now displays self-regulatory arrows with consistent positions regardless of the node size, edge thickness, or whether it is weighted. The GRNsight layout depends on the force graph engine from D3. We updated GRNsight from D3 version 3 to version 4, allowing the developers to utilize new D3 functionalities. We have implemented block layout where the nodes align into a grid alphabetically by node label. This graph layout feature will allow the users to organize and align the nodes with a simple button interface instead of manually dragging them as before. This will facilitate easier side-by-side comparison of the data from different GRN models. Alternatively, we plan to explore a hierarchical layout option as well. GRNsight is freely available at http://dondi.github.io/GRNsight/; the code is available under the open source BSD license at https://github.com/dondi/GRNsight.<br><br>
            
                                </p>
                                        
                                <p class="abstract">
                                    <b>Department of Biology, Loyola Marymount University, Los Angeles, California</b><br><br>
                                    September 2018<br>
                                    From the <i>lac</i> operon to the spaghettiome, what can we learn from modeling gene regulatory networks on a medium scale?<br>
                                    <b>Kam D. Dahlquist</b><br><br>
                                </p>
   
                                <p class="abstract">
                                    <b>Quantitative and Computational Biosciences Seminar, UCLA</b><br><br>
                                    March 2018<br>
                                    Dynamical Systems Modeling and Visualization of Gene Regulatory Networks: What Can We Learn from Networks on the “Medium” Scale?<br>
                                    <b>Kam D. Dahlquist</b><br><br>
                                </p>

                                <p class="abstract">
                                    <b>Frank R. Seaver College Professorial Lecture, Loyola Marymount University, Los Angeles, California</b><br><br>
                                    February 2018<br>
                                    The Process is the Product:  Systems Biology within an Open Science Ecosystem<br>
                                    <b>Kam D. Dahlquist</b><br><br>
                                </p>
                                        
                                <p class="abstract">
                                    <b>National Center for Ecological Analysis and Synthesis, Santa Barbara, California</b><br><br>
                                    March 2017<br>
                                    GRNmap and GRNsight:  Open Source Software for Dynamical Systems Modeling and Visualization of Medium-Scale Gene Regulatory Networks<br>
                                    <b>Kam D. Dahlquist</b><br><br>
                                </p>
                                        
                                <p class="abstract">
                                    <b>LMU Department of Biology Seminar Series</b><br><br>
                                    September 16, 2016<br>
                                    GRNmap and GRNsight: Using the power of genomics, mathematics,and open source visualization software to understand gene regulatory networks in yeast<br>
                                    <b>Kam D. Dahlquist</b><br><br>
                                </p>
<br>
                                
                                <p class="abstract">
                                    <b><a href="https://www.open-bio.org/wiki/BOSC_2016">Bioinformatics Open Source Conference (BOSC) 2016, Orlando, FL</a></b><br><br>
                                    July 9, 2016<br>
                                    GRNmap and GRNsight: open source software for dynamical systems modeling and visualization of medium-scale gene regulatory networks<br>
                                    <b>Kam D. Dahlquist</b>, Ben G. Fitzpatrick, John David N. Dionisio, Nicole A. Anguiano, Juan S. Carrillo, Trixie Anne M. Roque, Anindita Varshneya, Mihir Samdarshi, and Chukwuemeka Azinge<br><br>

                                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. Over a period of several years, our group has developed a MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program uses a penalized least squares approach (Dahlquist et al. 2015, DOI: 10.1007/s11538-015-0092-6) to estimate production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data, and then performs a forward simulation of the dynamics of the network. GRNmap has options for using a sigmoidal or Michaelis-Menten production function.  The large number of developers and time span of development led to a code base that was difficult to revise and adjust. We therefore brought the code under version control in a GitHub repository and refactored the script-based software with global variables into a function-based package that uses an object to carry relevant information from function to function. This modular approach allows for cleaner, less ambiguous code and increased maintainability. We standardized the format of the input and output Excel workbooks, making them more readable. We also added an optimization diagnostics output worksheet which includes both the actual and theoretical minimum least squared error overall, and the mean squared errors for the individual genes. The MATLAB compiler was used to create an executable that can be run on any Windows machine without the need of a MATLAB license, increasing the accessibility of our program. Finally, we have implemented test-driven development, creating unit tests for all new features to speed up debugging and to prevent future code regressions.  We are improving the test coverage of previous code.<br><br>
                                    GRNsight is an open source web application for visualizing such models of gene regulatory networks. GRNsight accepts GRNmap- or user-generated spreadsheets containing an adjacency matrix representation of the GRN and automatically lays out the graph of the GRN model. It is written in JavaScript, with diagrams facilitated by D3.js. Node.js and the Express framework handle server-side functions. GRNsight’s diagrams are based on D3.js’s force graph layout algorithm, which was then extensively customized. GRNsight uses pointed and blunt arrowheads, and colors the edges and adjusts their thicknesses based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. Visualizations can be modified through manual node dragging and sliders that adjust the force graph parameters. From the early stages, GRNsight has had a unit testing framework using Mocha and the Chai assertion library to perform test-driven development where unit tests are written before new functionality is coded. This framework consists of over 160 automated unit tests that examine over 450 test files to ensure that the program is running as expected. Error and warning messages inform the user what happened, the source of the problem, and possible solutions. <br><br>
                                    Together, the life cycle of these two programs illustrate the differences between the cultures of mathematics and computing, the challenges and benefits of bringing an existing code base up to open development standards (GRNmap), and the advantages of starting a project using best practices from the beginning (GRNsight).  Our goal is to facilitate reproducible research.<br><br>
                                     <a href="http://f1000research.com/slides/5-1637">Click here to view the slides on F1000Research.</a><br><br>
                                     
                                     <iframe src="//www.slideshare.net/GRNsight/slideshelf" width="615px" height="470px" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:none;" allowfullscreen webkitallowfullscreen mozallowfullscreen></iframe>
                                     
                                     <!--<figcaption><a href="http://www.slideshare.net/GRNsight/dahlquist-bosc-20160709-64224474" target="_blank">Slides</a> from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>-->
                                </p>
<br>
                                <p class="abstract">
                                    <b>Experimental Biology 2016, San Diego, CA</b><br><br>
                                    April 4, 2016<br>
                                    <a href="https://faseb.onlinelibrary.wiley.com/doi/10.1096/fasebj.30.1_supplement.819.16">GRNmap and GRNsight: open source software for dynamical systems modeling and visualization of medium-scale gene regulatory networks</a><br>
                                    <b>Kam D. Dahlquist</b>, Ben G. Fitzpatrick, John David N. Dionisio, Nicole A. Anguiano, Juan S. Carrillo, Monica V. Hong, Kristen M. Horstmann, Kayla C. Jackson, K. Grace Johnson, Tessa A. Morris, Trixie Anne M. Roque, Mihir Samdarshi, Anindita Varshneya, Natalie E. Williams, and Kevin W. Wyllie<br><br>

                                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. Our group has developed a MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program uses a penalized least squares approach (Dahlquist et al. 2015, DOI: 10.1007/s11538-015-0092-6) to estimate production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data, and then performs a forward simulation of the dynamics of the network. GRNmap has options for using a sigmoidal or Michaelis-Menten production function. Parameters for a series of related networks, ranging in size from 15 to 35 genes, were optimized against DNA microarray data measuring the transcriptional response to cold shock in wild type and five strains individually deleted for the transcription factors, Cin5, Gln3, Hap4, Hmo1, Zap1, of budding yeast, Saccharomyces cerevisiae BY4741. Model predictions fit the experimental data well, within the 95% confidence interval. Open source code and a compiled executable that can run without a MATLAB license are available from <a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>. GRNsight is an open source web application for visualizing such models of gene regulatory networks. GRNsight accepts GRNmap- or user-generated spreadsheets containing an adjacency matrix representation of the GRN and automatically lays out the graph of the GRN model. The application colors the edges and adjusts their thicknesses based on the sign (activation or repression) and the strength (magnitude) of the regulatory relationship, respectively. Users can then modify the graph to define the best visual layout for the network. The GRNsight open source code and application are available from <a href="http://dondi.github.io/GRNsight/index.html">http://dondi.github.io/GRNsight/index.html</a>.
                                </p>

                                <figure class="publication">
                                    <iframe src="//www.slideshare.net/slideshow/embed_code/key/NC3auEnK2HuL1f" width="595" height="485" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:1px solid #CCC; border-width:1px; margin-bottom:5px; max-width: 100%;" allowfullscreen></iframe>

                                    <figcaption><a href="//www.slideshare.net/GRNsight/dahlquist-experimental-biology20160404" title="Dahlquist experimental biology_20160404" target="_blank">Presentation slides</a> from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>

                                <p class="abstract">
                                    <b>5<sup>th</sup> Annual Southern California Systems Biology Conference</b><br><br>
                                    January 31, 2015<br>
                                    GRNmap and GRNsight: Open Source Software for Dynamical Systems Modeling and Visualization of Medium-Scale Gene Regulatory Networks<br>
                                    <b>Kam D. Dahlquist</b>,  Ben G. Fitzpatrick, John David N. Dionisio, Nicole A. Anguiano, Juan S. Carrillo, Nicholas A. Rohacz, Katrina Sherbina, Britain J. Southwick, and Anindita Varshneya<br><br>

                                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. Our group has developed a MATLAB software package, called GRNmap, that uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program uses a penalized least squares approach to estimate production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data, and then performs a forward simulation of the dynamics of the network. Parameters for a 21-gene network were optimized against DNA microarray data measuring the transcriptional response to cold shock in wild type and four transcription factor deletion strains of budding yeast, Saccharomyces cerevisiae. Model predictions fit experimental data well, within the 95% confidence interval. Open source code and a compiled executable that can run without a MATLAB license are available from <a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>. GRNsight is an open source web application for visualizing such models of gene regulatory networks. GRNsight accepts GRNmap- or user-generated spreadsheets containing an adjacency matrix representation of the GRN and automatically lays out the graph of the GRN model. The application colors the edges and adjusts their thicknesses based on the sign (activation or repression) and the strength (magnitude) of the regulatory relationship, respectively. Users can then modify the graph to define the best visual layout for the network. The GRNsight code and application are available from <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/index.html</a>. This work was partially supported by NSF award 0921038.
                                </p>

                                <figure class="publication">
                                    <iframe src="//www.slideshare.net/slideshow/embed_code/44236648" width="425" height="355" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:1px solid #CCC; border-width:1px; margin-bottom:5px; max-width: 100%;" allowfullscreen></iframe>
                                    <figcaption><a href="http://www.slideshare.net/GRNsight/dahlquist-so-calsysbio20140131" title="Dahlquist so calsysbio_20140131" target="_blank">Presentation slides</a> from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>

                                <p class="abstract">
                                    <b>Loyola Marymount University CMSI 402 Presentation</b><br><br>
                                    May 8, 2014<br>
                                    GRNsight: a Web Application for Visualizing Models of Gene Regulatory Networks<br>
                                    <b>Britain Southwick</b>
                                </p>

                                <figure class="publication">
                                    <iframe src="//www.slideshare.net/slideshow/embed_code/35811802" width="427" height="356" allowfullscreen></iframe>

                                    <figcaption><a href="https://www.slideshare.net/GRNsight/southwick-britain-grnsightcmsi402presentation20140508" title="Southwick britain gr_nsight_cmsi402-presentation_20140508" target="_blank">Presentation slides</a> from <a href="http://www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>

                                <p class="abstract">
                                    <b>Loyola Marymount University 6<sup>th</sup> Annual Undergraduate Research Symposium</b><br><br>
                                    March 29, 2014<br>
                                    GRNsight: a Web Application for Visualizing Models of Gene Regulatory Networks<br>
                                    <b>Britain Southwick, Nicole Anguiano</b><br>
                                    John David N. Dionisio, Kam D. Dahlquist, Ben G. Fitzpatrick<br><br>

                                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them, which govern the level of expression of mRNA and protein from those genes. GRNs can be mathematically modeled and simulated by applications such as GRNmap, a MATLAB program that estimates the parameters and performs forward simulations of a differential equations model of a GRN. Computer representations of GRNs, such as the models output by GRNmap, are tabular spreadsheets that are not easily interpretable. Ideally, GRNs should be displayed as diagrams (graphs) detailing the regulatory relationships (edges) between each gene (node) in the network. To address this need, we developed GRNsight. GRNsight is an open source web application for visualizing mathematical models of GRNs. It allows users to upload spreadsheets generated by GRNmap and uses the information in these spreadsheets to automatically create and display a graph of the GRN model. The application colors the edges and adjusts their thickness based on the sign (activation or repression) and the strength (magnitude) of the regulatory relationship, respectively. Finally, GRNsight then allows the user to modify the graph in order to define the best visual layout for the network. Most of GRNsight is written in JavaScript. HTTP requests are handled using Node.js and the Express framework. Graphs are generated through D3.js, a JavaScript data visualization library. <br>
                                </p>

                                <figure class="publication">
                                    <iframe src="//www.slideshare.net/slideshow/embed_code/35811669" width="427" height="356" allowfullscreen></iframe>
                                    <figcaption><a href="https://www.slideshare.net/GRNsight/southwick-anguiano-lmusymposiumpresentation20140329" title="Southwick anguiano lmu-symposium_presentation_20140329" target="_blank">Presentation slides</a> from <a href="http://www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>
                            </div>
                        </div>
                    </div>

                    <div class="panel panel-default">
                        <div class="panel-heading">
                            <p class="panel-title aboutHeaders">
                                <a href="#posters" data-toggle="collapse">
                                    Posters
                                </a>
                            </p>
                        </div>

                        <div id="posters" class="panel-collapse collapse">
                            <div class="panel-body">
                                <p class="abstract">
                    <b>Institute for Quantitative and Computational Biology Retreat, University of Californina, Los Angeles</a></b><br><br>
                    <b>September 22, 2025</b><br>
                    Sensitivity analysis of GRNmap and new features for GRNsight: open source software for dynamical systems modeling and visualization of small-scale gene regulatory networks<br>
                    <b>Kam D. Dahlquist</b>, Ben G. Fitzpatrick, John David N. Dionisio, Ngoc K. Tran, Nikki C. Chun, Cecilia J. Zaragoza, A'Kaia L. Phelps, Amelie T. Dinh, Milka Y. Zekarias<br><br>
                    <b>Abstract</b><br>
                    A gene regulatory network (GRN) is a set of transcription factors that regulate the expression of genes encoding other transcription factors. GRNmap is an open source MATLAB package that uses ordinary differential equations to model dynamics of small-scale GRNs. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data and then performs forward simulations of model dynamics. While the model has been successfully used to understand networks of 15-20 genes, we wanted to closely examine how it works on a smaller scale to determine parameter sensitivity. All 21 possible 'toy' networks of 3 nodes and 4 edges were created based on the simulated expression data outputted when known parameters were run through a forward simulation. Then the simulated data was used to estimate the parameters again. Comparison of the known to estimated parameters showed that estimating production rates in addition to weights and thresholds reduced the accuracy of the results. The model was also sensitive to the direction and magnitude of the weight parameters for networks with the same connectivity. GRNsight is an open-source web application for visualizing such models of GRNs, automatically displaying a graph where edges are color-coded based on the activation and repression relationships, and nodes are color-coded with time course gene expression data. Backend gene expression and gene regulatory network databases allow users to generate a network, color the notes with expression data, and export an Excel workbook for modeling in GRNmap.<br>

                        <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/ISMBECCB2025_1759_Dahlquist_poster.pdf"> Link to poster (also presented at IMSB 2025 below)<a/> <br>
                <br>
                </p>                                    
                                        
                                <p class="abstract">
                    <b><a href="https://www.iscb.org/ismbeccb2025/home">ISMB/ECCB 2025, Liverpool, England</a></b><br><br>
                    <b>July 20-25, 2025</b><br>
                    Sensitivity analysis of GRNmap and new features for GRNsight: open source software for dynamical systems modeling and visualization of small-scale gene regulatory networks<br>
                    <b>Kam D. Dahlquist</b>, Ben G. Fitzpatrick, John David N. Dionisio, Ngoc K. Tran, Nikki C. Chun, Cecilia J. Zaragoza, A'Kaia L. Phelps, Amelie T. Dinh, Milka Y. Zekarias<br><br>
                    <b>Abstract</b><br>
                    A gene regulatory network (GRN) is a set of transcription factors that regulate the expression of genes encoding other transcription factors. GRNmap is an open source MATLAB package that uses ordinary differential equations to model dynamics of small-scale GRNs. The program estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data and then performs forward simulations of model dynamics. While the model has been successfully used to understand networks of 15-20 genes, we wanted to closely examine how it works on a smaller scale to determine parameter sensitivity. All 21 possible 'toy' networks of 3 nodes and 4 edges were created based on the simulated expression data outputted when known parameters were run through a forward simulation. Then the simulated data was used to estimate the parameters again. Comparison of the known to estimated parameters showed that estimating production rates in addition to weights and thresholds reduced the accuracy of the results. The model was also sensitive to the direction and magnitude of the weight parameters for networks with the same connectivity. GRNsight is an open-source web application for visualizing such models of GRNs, automatically displaying a graph where edges are color-coded based on the activation and repression relationships, and nodes are color-coded with time course gene expression data. Backend gene expression and gene regulatory network databases allow users to generate a network, color the notes with expression data, and export an Excel workbook for modeling in GRNmap.<br>

                        <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/ISMBECCB2025_1759_Dahlquist_poster.pdf"> Link to poster<a/> <br>
                        <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/ISMBECCB2025_1759_Dahlquist_poster.mp4"> Link to video about poster</a> 
                <br>
                </p>                                      
                                    
                               <p class="abstract">
                                    <b>Loyola Marymount University 17<sup>th</sup> Annual Undergraduate Research Symposium</a></b><br><br>
                                    <b>March 21, 2025</b><br>
                                    Database, Display, and DevOps Improvements for GRNsight 7.2, a Web Application for Visualizing Gene Regulatory and Protein-Protein Interaction Network Models <br>
                                    <b>Ngoc K. Tran, Cecilia J. Zaragoza, A'Kaia Phelps, Amelie Dinh, Milka Zekarias</b>, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is an open-source web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them, which govern the expression level of mRNA and protein from genes. GRNsight can now also display protein-protein physical interaction (PPI) networks. Whereas GRNs have directed edges, PPI networks have undirected edges to indicate protein binding. We have continued to integrate PPI networks so that users can display node coloring on a PPI with a choice of top or bottom datasets of expression data. GRNsight reads user-uploaded Microsoft Excel workbooks and automatically displays a graph. Users without their own data can use GRNsight's back-end PostgreSQL database to select a GRN or PPI based on yeast data from AllianceMine. Database loading scripts were ported to AllianceMine because YeastMine was discontinued. New graph functionality has been implemented that allows users to restrict the graph to viewport so that they can drag, zoom, and move the graph with the nodes staying within the application’s bounding box. On the DevOps side, the project’s continuous integration/development pipeline has been migrated from Travis CI to GitHub Actions, integrating workflows for automated builds, unit tests, and linting to streamline development processes and improve maintainability. Ongoing efforts prioritize bug resolution, user interface enhancements, and improved documentation. GRNsight is now positioned to comprehensively address GRNs and PPIs, offering a unified platform for visualizing diverse molecular interaction models. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.<br><br>
                                     <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/Tran_Zaragoza_Phelps_Dinh_Zekarias_LMU_SYMPOSIUM_2025.pdf">Click here</a> to download the poster. <br><br>

                                </p>

                                <p class="abstract">
                                    <b>Loyola Marymount University 16<sup>th</sup> Annual Undergraduate Research Symposium</a></b><br><br>
                                    <b>March 15, 2024</b><br>
                                    Propagating Protein-Protein Interaction Network Support into GRNsight 7.0, a Web Application for Visualizing Gene Regulatory Network Models <br>
                                    <b>Ngoc K. Tran, Cecilia J. Zaragoza, A'Kaia Phelps</b>, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is an open-source web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them, which govern the expression level of mRNA and protein from genes. GRNsight reads user-uploaded Microsoft Excel adjacency matrices or SIF files and automatically displays a graph. Users without their own data can use GRNsight's back-end PostgreSQL database to select a GRN based on the Saccharomyces Genome Database (SGD). Besides GRN data, we have recently incorporated protein-protein physical interaction (PPI) data from SGD so that users can also visualize this type of network. Due to this new type of network, the rest of the application needed updates in order to fully support PPIs. For example, PPI networks have undirected edges vs. directed edges for GRNs. Gene and protein labels also differ. Notable additions include a PPI demo graph, allowing users to visualize a graph automatically without inputting specific data. This new demo exclusively showcases interactions involving yeast mitochondrial protein Aim32p, offering users a detailed exploration of its network connections. Furthermore, GRNsight now incorporates functionality to detect the network type of the imported file, distinguishing between GRNs and PPIs. Ongoing development efforts prioritize bug resolution, user interface enhancements, and improved documentation. In response to user needs, GRNsight is now positioned to comprehensively address GRNs and PPIs, offering a unified platform for visualizing diverse molecular interaction models. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.<br><br>
                                     <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/Tran_Zaragoza_Phelps_LMU_SYMPOSIUM_2024.pdf">Click here</a> to download the poster. <br><br>
                                    </p>

                                <p class="abstract">
                                    <b><a href="https://genetics-gsa.org/tagc-2024/">The Allied Genetics Conference (TAGC24), Metro Washington DC</a></b><br><br>
                                    <b>March 6-10, 2024</b><br>
                                    Extending the GRNsight application for visualizing small-to-medium gene regulatory networks to incorporate physical protein-protein interaction data from SGD <br>
                                    <b>Kam D. Dahlquist</b>, Onariaginosa O. Igbinedion, Ahmad R. Mersaghian, Ngan N. Tran, Deepa V. Dabir, John David N. Dionisio<br><br>

                                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap is an open source MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN based on time course gene expression data. GRNsight is an open source web application for visualizing small-to-medium scale GRNs, especially models produced by GRNmap. GRNsight reads the Excel input and output workbooks from GRNmap and automatically displays the model data as a graph with colored nodes (expression data) and edges (estimated regulatory weights). We recently implemented a backend PostgreSQL database for GRNsight, populated with five public gene expression datasets and regulatory network data from the Saccharomyces Genome Database (SGD). A user can select genes to include in the GRN, and GRNsight will automatically layout the network using regulatory connections from SGD. The user can then select one of the time course gene expression datasets with which to color the nodes and export to a properly-formatted GRNmap input workbook. GRNmap can then be used to estimate the regulatory weights (activation vs. repression and magnitude of the relationship). Although GRNsight was originally designed for GRNs, it has the general applicability for displaying any appropriately-sized, weighted or unweighted network with directed edges. Previously, GRNsight was limited to only displaying graphs with directed edges and was unable to display protein-protein physical interaction (PPI) networks, as they are represented by undirected graphs. Thus a protein-protein physical interaction network mode was created to visualize these models in GRNsight. Additionally, a backend database was added to store PPI data obtained from SGD, so that users can generate a PPI network within GRNsight the same way they can generate a GRN. The graphical user interface was updated to be able to toggle between these two graph modes, enabling comparison of these two types of data. GRNmap is available at <a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>; GRNsight is available at <a href="http://dondi.github.io/GRNsight">http://dondi.github.io/GRNsight</a>.<br><br>
                                     <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/Dahlquist-et-al_TAGC_2024.pdf">Click here</a> to download the poster. <br><br>                                 
                                </p>

                                <p class="abstract">
                                    <b>Loyola Marymount University 15<sup>th</sup> Annual Undergraduate Research Symposium</b><br><br>
                                    <b>March 17, 2023</b><br>
                                    Improved Functionality of GRNsight 6.0: a Web Application for Visualizing Gene Regulatory Network Models <br>
                                    <b>Ahmad R. Mersaghian, Sarron A. Tadesse, Ngan N. Tran</b>, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is an open-source web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap is a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN based on user-provided expression data. GRNsight reads Microsoft Excel input and output workbooks from GRNmap and automatically displays the model data as a graph. Graph edges are color-coded based on the activation and repression relationships between the transcription factors, and nodes are color-coded with time course gene expression data. The addition of backend gene expression and gene regulatory network databases allows users with no data of their own to generate a network, color the notes with expression data, and export an Excel workbook for modeling in GRNmap. Network data was obtained from the Saccharomyces Genome Database via the Yeastmine search and retrieval interface. A new gene expression dataset from Apweiler et al. (2012) was integrated into the GRNsight database and user interface. Developer documentation for the databases was improved, and the Application Programming Interface (API) and Data Access Layer (DAL) modules were refactored and expanded to improve separation of concerns. In parallel, the ongoing development of GRNsight is focused on fixing bugs and adding enhancements to the user interface and graph layout. GRNsight is freely available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.<br><br>
                                     <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/Mersaghian_Tadesse_Tran_LMU_SYMPOSIUM_2023.pdf">Click here</a> to download the poster. <br><br>
                                    </p>

                                    <p class="abstract">
                                    <b><a href="https://genetics-gsa.org/yeast-2022/">Yeast Genetics Meeting, University of California, Los Angeles</a></b><br><br>
                                    <b>August 17–21, 2022</b><br>
                                    Integration of SGD regulatory and expression data into the GRNmap and GRNsight applications for modeling and visualizing small-to-medium gene regulatory networks <br>
                                    <b>Kam D. Dahlquist</b>, Onariaginosa O. Igbinedion, Ahmad R. Mersaghian, Sarron A. Tadesse, John David N. Dionisio<br><br>

                                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap is an open source MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN based on time course gene expression data. GRNsight is an open source web application for visualizing small-to-medium scale GRNs, especially models produced by GRNmap. GRNsight reads the Excel input and output workbooks from GRNmap and automatically displays the model data as a graph with colored nodes (expression data) and edges (estimated regulatory weights). A limitation for GRNmap has been the manual creation of input Excel workbooks which is time-consuming, error-prone, and dependent upon the user having their own network and expression data. To address this, we have implemented a backend PostgreSQL database for GRNsight, populated with five public gene expression datasets (Apweiler et al. 2012, GSE33098; Barreto et al. 2012, GSE24712; Dahlquist et al. 2018, GSE83656; Kitagawa et al. 2002, GSE9336; Thorsen et al. 2007, GSE6068) and regulatory network data from the Saccharomyces Genome Database (SGD). A user can now select genes to include in the GRN, and GRNsight will automatically layout the network using regulatory connections from SGD. The user can then select one of the time course gene expression datasets with which to color the nodes and export to a properly-formatted GRNmap input workbook. GRNmap can then be used to estimate the regulatory weights (activation vs. repression and magnitude of the relationship). A GRNmap executable is available for users who do not have access to a MATLAB license. Finally, the GRNmap output can be loaded back into GRNsight to visualize the results. Closing this loop for automating and validating the creation of GRNmap input workbooks speeds up the rate of research, enabling the comparison of models of different GRNs with the same expression data source or the same GRN with different expression data sources. GRNmap is available at <a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>; GRNsight is available at <a href="http://dondi.github.io/GRNsight">http://dondi.github.io/GRNsight</a>.<br><br>
                                     <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/Dahlquist-et-al_Yeast-Genetics_2022.pdf">Click here</a> to download the poster. <br><br>
                                    </p>

<p class="abstract">
                                    <b>West Coast Biological Sciences Undergraduate Research Conference</b><br><br>
                                    <b>April 9, 2022</b><br>
                                    A New Gene Expression Dataset for GRNsight: a Web Application for Visualizing Gene Network Models<br>
                                    <b>Ahmad R. Mersaghian</b>, Sarron A. Tadesse, John David N. Dionisio, Kam D. Dahlquist<br><br>
                                    Abstract: GRNsight is an open-source web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap is a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN based on user-provided expression data. GRNsight reads Microsoft Excel input and output workbooks from GRNmap and automatically displays the model data as a graph. Graph edges are color-coded based on the activation and repression relationships between the transcription factors, and nodes are color-coded with time course gene expression data provided by the user or retrieved from the backend database. To expand the potential research scope, a new expression dataset was chosen to be added to the GRNsight database. Apweiler et al. (2012) measured gene expression changes over time in Saccharomyces cerevisiae due to adding glucose to the media. To clean this dataset for inclusion in the database, we needed to remove duplicate data and assign the correct ID types. To further analyze the data with GRNmap, we clustered with STEM to identify significant expression profiles and inferred the gene regulatory network with YEASTRACT. After running GRNmap, we visualized the results of the model using GRNsight. In parallel, ongoing development of GRNsight is focused on fixing bugs and adding enhancements to the user interface and graph layout. GRNsight is freely available at <a href="http://dondi.github.io/GRNsight">http://dondi.github.io/GRNsight</a>.<br><br>

                                     <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/Mersaghian_Tadesse_LMU_WCBSURC_2022.pdf">Click here</a> to download the poster. <br><br>
                                    </p>

<p class="abstract">
                                    <b>Loyola Marymount University 14<sup>th</sup> Annual Undergraduate Research Symposium</b><br><br>
                                    <b>March 18, 2022</b><br>

                                     New Databases and Export Features for GRNsight: a Web Application for Visualizing Models of GRNs<br>
                                    <b>Onariaginosa O. Igbinedion</b>, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is an open source web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap is a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN based on user-provided expression data. A limitation for GRNmap is the manual creation of input Excel workbooks which is time-consuming and can be error-prone. GRNsight reads these input and output workbooks from GRNmap and automatically displays the model data as a graph with colored nodes and edges. We have updated the existing back-end GRNsight database to be compatible with other species in addition to Saccharomyces cerevisiae. We cleaned up the database schema and added appropriate types to the data so that proper sorting is achieved when retrieving expression data. We expanded the user interface for GRNsight’s export-to-Excel functionality and connected it to the database, so that users can select which particular expression data sheets to export. Furthermore, we are storing all of the interactions between regulatory genes in the database so that GRNsight users can specify which genes belong in their GRN of interest. In conjunction with the updated expression data export feature, this will allow users to fully automate and validate the creation of GRNmap input workbooks, speeding up the rate of research. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.<br><br>
                                     <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/Igbinedion_LMU_SYMPOSIUM_2022.pdf">Click here</a> to download the poster. <br><br>

            
                                    A New Gene Expression Dataset for GRNsight: a Web Application for Visualizing Gene Network Models<br>
                                    <b>Ahmad R. Mersaghian, Sarron A. Tadesse</b>, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is an open-source web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap is a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN based on user-provided expression data. GRNsight reads Microsoft Excel input and output workbooks from GRNmap and automatically displays the model data as a graph. Graph edges are color-coded based on the activation and repression relationships between the transcription factors, and nodes are color-coded with time course gene expression data provided by the user or retrieved from the backend database. To expand the potential research scope, a new expression dataset was chosen to be added to the GRNsight database. Apweiler et al. (2012) measured gene expression changes over time in Saccharomyces cerevisiae due to adding glucose to the media. To clean this dataset for inclusion in the database, we needed to remove duplicate data and assign the correct ID types. To further analyze the data with GRNmap, we clustered with STEM to identify significant expression profiles and inferred the gene regulatory network with YEASTRACT. After running GRNmap, we visualized the results of the model using GRNsight. In parallel, ongoing development of GRNsight is focused on fixing bugs and adding enhancements to the user interface and graph layout. GRNsight is freely available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.<br><br>
                                     <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/Mersaghian_Tadesse_LMU_SYMPOSIUM_2022.pdf">Click here</a> to download the poster. <br><br>

                                </p>
                                        
                                    <p class="abstract">
                                    <b>Loyola Marymount University 13<sup>th</sup> Annual Undergraduate Research Symposium (virtual)</b><br><br>
                                    <b>March 19-20, 2021</b><br>
                                    More Robust Testing of Data and UI for GRNsight: a Web Application for Visualizing Models of GRNs <br>
                                    <b>Onariaginosa O. Igbinedion, Ian M. Green, Ahmad R. Mersaghian</b>, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is an open source web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN, can mathematically model GRNs. GRNsight reads Microsoft Excel input and output workbooks from GRNmap and automatically displays the model data as a graph. Graph edges are color-coded based on the activation and repression relationships between the transcription factors, and nodes are color-coded with time course gene expression data. A new export-to-Excel function enables GRNsight to be used as a input workbook format validator for GRNmap, which GRNmap lacks. Toward this end, we improved the import and export functionality of GRNsight by adding 28 new tests to the testing suite, while simultaneously consolidating and debugging existing tests. This increased user awareness of specific errors and warnings in the import and export process, allowing them to fix problems with Excel workbooks before attempting to run GRNmap. Additionally, we revised the User Interface (UI) to fix graph edge display bugs, as well as viewport selection and movement issues. Along the way, we updated and streamlined the testing documentation for conducting further manual UI tests. Finally, we updated antiquated code libraries to take advantage of new features. GRNsight is freely available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.<br><br>
                                     <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/Igbinedion_Green_Mersaghian_LMU_SYMPOSIUM_2021.pdf">Click here</a> to download the poster.<br><br>

<p class="abstract">
                                    <b>11<sup>th</sup> Annual Loyola Marymount University Undergraduate Research Conference</b><br><br>
                                    March 2019<br>
                                    A Dynamic Gene Page Feature for GRNsight: a Web Application for Visualizing Models of Gene Regulatory Networks<br>
                                    <b>John L. Lopez, Alexia M. Filler</b>, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN, can mathematically model GRNs. Using a GRNmap output workbook, GRNsight automatically lays out the network graph through a force simulation or arranges the nodes in a grid. GRNsight uses pointed and blunt arrowheads on the edges and color-codes and adjusts edge thicknesses based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. Nodes are colored based on experimental or simulated time course gene expression data found in the GRNmap workbook. We have also added a feature that gives users the option to export data to an GRNmap-formatted Excel workbook. Additionally, we implemented a dynamic gene information webpage. When a user right-clicks on a node in the graph, a pop-up webpage sends a series of data requests to the JASPAR, NCBI Gene, UniProt, Ensembl, and Saccharomyces Genome Databases. The webpage is populated with data returned regarding the gene’s description, protein sequence, DNA binding motif frequency matrix and sequence logo, and Gene Ontology terms. This new feature enables users to access information about the genes to facilitate biological interpretation of the GRN data, and is available in GRNsight version 3.1.0 at <a href="http://dondi.github.io/GRNsight">http://dondi.github.io/GRNsight</a>.<br>

                                            Click <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/Lopez_Filler_LMU_Symposium_2019.pdf" target="_blank">here</a> to view the slides. 
                                </p>
                                                
                                <p class="abstract">
                                    <b>8<sup>th</sup> Annual Southern California Systems Biology Conference</b><br><br>
                                    February 2019<br>
                                    New Layouts, Data Types, and Architecture for GRNsight 3:  a Web Application for Visualizing Gene Regulatory Networks<br>
                                    <b>Mihir Samdarshi</b>, John L. Lopez, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing small- to medium-scale gene regulatory networks (GRNs). A GRN consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNsight accepts input files in the Excel, SIF, and GraphML formats and then produces weighted or unweighted network graphs by representing genes as nodes and regulatory connections as edges. Activation is represented with red edges, repression is represented with blue edges, and weight magnitudes below a certain threshold are represented with gray edges to denote weak activation or repression relationships. GRNsight now contains features to facilitate network comparison. Nodes on each graph show a heat map corresponding to gene expression levels over time; these nodes can be overlaid with up to two different data sets. The graphs generated by GRNsight may now be automatically laid out in a grid. When a user right-clicks on a gene, data requests are made to five public databases and a web page is dynamically populated with the data returned. Finally, a number of under the hood improvements were made to GRNsight’s architecture. Most notably, GRNsight was made to follow a strict Model-View-Controller (MVC) paradigm. MVC consolidates the application’s state into a single object, from which presentation and interaction are derived. GRNsight version 3.1.3 is freely available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>; the code is available under the open source BSD license at <a href="https://github.com/dondi/GRNsight">https://github.com/dondi/GRNsight</a>.<br>
                                            Click <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/Samdarshi-et-al_SoCalSysBio_2019.pdf" target="_blank">here</a> to view the poster. 
                                </p>

                                <p class="abstract">
                                    <b>Yeast Genetics Meeting, Stanford University, California</b><br><br>
                                    August 2018<br>
                                    Mathematical modeling of small gene regulatory networks reveals key regulators and network properties important for controlling the early response to cold shock in <i>Saccharomyces cerevisiae</i><br>
                                    <b>Kam D. Dahlquist</b>, Ben G. Fitzpatrick, Brandon J. Klein, Margaret J. O’Neil, Lauren M. Kelly<br><br>

                                    Abstract: A gene regulatory network (GRN) is a group of transcription factors that control the level of expression of genes encoding other transcription factors. Dynamics of GRNs illustrate how expression in the network changes over time. GRNmap, a MATLAB software package (http://kdahlquist.github.io/GRNmap/), uses differential equations to model the dynamics of small-scale GRNs. The software estimates production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on microarray data. Microarray data was obtained from a cold shock experiment where wild-type budding yeast, Saccharomyces cerevisiae, and five strains from which the transcription factors Cin5, Gln3, Hap4, Hmo1, and Zap1 were individually deleted were subjected to cold shock at 13°C for 15, 30, and 60 minutes. Six related GRNs, which ranged from 14-17 transcription factors (nodes) and 25-35 regulatory relationships (edges), were constructed using the microarray data and the YEASTRACT database. GRNmap was then used to estimate production rates, expression thresholds, and regulatory weights for each of these GRNs, followed by visualization of the results with the GRNsight software (http://dondi.github.io/GRNsight/). Forward simulation of the model showed a good fit to the experimental data, particularly in comparison to 30 random networks with the same genes and number of edges. Systematic analysis of edges revealed repeated motifs in the database-derived networks. These motifs included a coherent type I feed forward loop (FFL), an activating regulatory chain, and symmetrical incoherent type I FFLs terminating on the paralogs Yhp1 and Yox1. Multiple regression analysis of the database-derived and random networks demonstrated that model fit to experimental expression data was repeatedly correlated with the eigenvector centrality of individual transcription factors. Better model fit to expression data was observed for transcription factors with high eigenvector centrality, such as Gcn4 and Yhp1. A consolidated network containing 15 transcription factors and 34 edges was constructed from conserved motifs and high eigenvector centrality transcription factors. The consolidated network outperformed all random networks and all but one database-derived network. The consolidated network also featured appreciable overlap with the general environmental stress response, with Hmo1 activating Msn2/Msn4 directly and Yhp1 indirectly.br>
                                            Click <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/Dahlquist-et-al_yeast-genetics-poster_20180821.pdf" target="_blank">here</a> to view the poster. 
                                </p>
                                        
                                <p class="abstract">
                                    <b>Beta Beta Beta Biological Honor Society's Pacific District Convention, Concordia University, Irvine, California</b><br><br>
                                    March 2018<br><br>
                                    Data Comparison Features and Development Tool Improvements for GRNsight: a Web App for Visualizing Gene Regulatory Networks<br>
                                    <b>Mihir Samdarshi</b>, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing small- to medium-scale gene regulatory networks (GRNs). A GRN consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNsight accepts input files in the Excel, SIF, and GraphML formats and then produces weighted or unweighted network graphs by representing genes as nodes and regulatory connections as edges. Activation is represented with magenta edges, repression is represented with cyan edges, and weight magnitudes below a certain threshold are represented with gray edges to denote weak activation or repression relationships. To better assess the relative strengths of the regulatory relationships, we added a slider that will adjust the gray threshold based on the magnitude of the weight. To facilitate comparison of graphs with different ranges of weight magnitudes, we added the ability to adjust weight normalization factors that control the thicknesses of the edges. Under the hood, the testing framework was refactored into semantic vs. syntactic tests of the validity of input files. Our development tool chain now also includes automatic test coverage calculation (currently 79%), linting for improved code style and readability, and continuous integration for quality assurance. Underlying third-party software libraries including Express, Node-XLSX, and D3 were updated to maintain compatibility and take advantage of new features. GRNsight is freely available at http://dondi.github.io/GRNsight/; the code is available under the open source BSD license at https://github.com/dondi/GRNsight.<br><br>
                                    Click <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/Samdarshi-et-al_LMU-Symposium_2018.pdf" target="_blank">here</a> to view the poster, which was also presented at the LMU Undergraduate Research Symposium. <b><i>Mihir was awarded Honorable Mention for the John C. Johnson Award for Excellence in Student Research for this poster presentation.</b></i> <br><br>

                                </p>
                                        
                                <p class="abstract">
                                    <b>10<sup>th</sup> Annual Loyola Marymount University Undergraduate Research Conference</b><br><br>
                                    March 2018<br><br>
                                    Data Comparison Features and Development Tool Improvements for GRNsight: a Web App for Visualizing Gene Regulatory Networks<br>
                                    <b>Mihir Samdarshi</b>, John David N. Dionisio, Kam D. Dahlquist<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing small- to medium-scale gene regulatory networks (GRNs). A GRN consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNsight accepts input files in the Excel, SIF, and GraphML formats and then produces weighted or unweighted network graphs by representing genes as nodes and regulatory connections as edges. Activation is represented with magenta edges, repression is represented with cyan edges, and weight magnitudes below a certain threshold are represented with gray edges to denote weak activation or repression relationships. To better assess the relative strengths of the regulatory relationships, we added a slider that will adjust the gray threshold based on the magnitude of the weight. To facilitate comparison of graphs with different ranges of weight magnitudes, we added the ability to adjust weight normalization factors that control the thicknesses of the edges. Under the hood, the testing framework was refactored into semantic vs. syntactic tests of the validity of input files. Our development tool chain now also includes automatic test coverage calculation (currently 79%), linting for improved code style and readability, and continuous integration for quality assurance. Underlying third-party software libraries including Express, Node-XLSX, and D3 were updated to maintain compatibility and take advantage of new features. GRNsight is freely available at http://dondi.github.io/GRNsight/; the code is available under the open source BSD license at https://github.com/dondi/GRNsight.<br><br>
                                    Click <a href="https://github.com/dondi/GRNsight-archive/blob/main/documents/posters/Samdarshi-et-al_LMU-Symposium_2018.pdf" target="_blank">here</a> to view the poster. <br><br>

                                </p>            
                            
                                <p class="abstract">
                                    <b><a href="https://www.open-bio.org/wiki/BOSC_2016">Bioinformatics Open Source Conference (BOSC) 2016</a> and <a href="http://www.iscb.org/ismb2016">Intelligent Systems for Molecular Biology (ISMB) 2016</a>, Orlando, FL</a></b><br><br>
                                    July 8-12, 2016<br>
                                    GRNmap and GRNsight: open source software for dynamical systems modeling and visualization of medium-scale gene regulatory networks<br>
                                    <b>Kam D. Dahlquist</b>, Ben G. Fitzpatrick, John David N. Dionisio, Nicole A. Anguiano, Juan S. Carrillo, Tessa A. Morris, Anindita Varshneya, Natalie E. Williams, K. Grace Johnson, Trixie Anne M. Roque, Kristen M. Horstmann, Mihir Samdarshi, Chukwuemeka E. Azinge, Brandon J. Klein, Margaret J. O’Neil1<br><br>

                                    Abstract: A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them that govern the level of expression of mRNA and proteins from those genes. Our open source MATLAB software package, GRNmap (<a href="http://kdahlquist.github.io/GRNmap/">http://kdahlquist.github.io/GRNmap/</a>), uses ordinary differential equations to model the dynamics of medium-scale GRNs. The program uses a penalized least squares approach (Dahlquist et al. 2015, DOI: 10.1007/s11538-015-0092-6) to estimate production rates, expression thresholds, and regulatory weights for each transcription factor in the network based on gene expression data, and then performs a forward simulation of the dynamics of the network. GRNmap has options for using a sigmoidal or Michaelis-Menten production function.  Parameters for a series of related networks, ranging in size from 15 to 35 genes, were optimized against DNA microarray data measuring the transcriptional response to cold shock in budding yeast, Saccharomyces cerevisiae, for the wild type strain and strains deleted for the transcription factors Cin5, Gln3, Hap4, Hmo1, and Zap1, giving biological insights into this process. GRNsight is an open source web application for visualizing such models of gene regulatory networks (<a href="http://dondi.github.io/GRNsight/index.html">http://dondi.github.io/GRNsight/index.html</a>). GRNsight accepts GRNmap- or user-generated Excel spreadsheets containing an adjacency matrix representation of the GRN and automatically lays out the graph. The application colors the edges and adjusts their thicknesses based on the sign (activation or repression) and the strength (magnitude) of the regulatory relationship, respectively. Users can then modify the graph to define the best visual layout for the network. This work was partially supported by NSF award 0921038.<br><br>
                                     <a href="http://f1000research.com/posters/5-1618">Click here</a> to view the poster on F1000Research. <a href="//www.slideshare.net/GRNsight/dahlquist-etal-boscismb2016poster-64224013" title="Dahlquist et-al bosc-ismb_2016_poster" target="_blank">Click here</a> to view poster from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare.<br><br>

                                </p>
                                <p class="abstract">
                                    <b>Experimental Biology 2016, San Diego, CA</b><br><br>
                                    April 4, 2016<br>
                                    <a href="https://faseb.onlinelibrary.wiley.com/doi/abs/10.1096/fasebj.30.1_supplement.819.10" target="_blank">Test-driven development improves GRNsight: a web application for visualizing models of gene regulatory networks</a><br>
                                    <b>Anindita Varshneya, Mihir Samdarshi</b>, Kam D. Dahlquist, John David N. Dionisio, and Ben G. Fitzpatrick<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN, can mathematically model the dynamics of GRNs. GRNsight automatically lays out the network graph based on GRNmap output spreadsheets. GRNsight uses pointed and blunt arrowheads, and colors the edges and adjusts their thicknesses based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. Visualizations can be modified through manual node dragging and sliders that adjust the force graph parameters. We have now implemented an exhaustive unit testing framework using Mocha and the Chai assertion library to perform test-driven development where unit tests are written before new functionality is coded. This framework consists of over 135 automated unit tests that examine about 450 test files to ensure that the program is running as expected. Error and warning messages have a three-part framework that informs the user what happened, the source of the problem, and possible solutions. For example, GRNsight returns an error when the spreadsheet is formatted incorrectly or the maximum number of nodes or edges is exceeded. The completion of the testing framework marks the close of development for version 1 (the current release stands at version 1.12). In version 2.0 of GRNsight, a new feature will be implemented that colors the nodes (genes) based on the expression data. The user will have the choice to display the experimental data or the data produced by the forward simulation feature in GRNmap. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.
                                </p>

                                <figure class="publication">
                                    <iframe src="//www.slideshare.net/slideshow/embed_code/key/neLFcr6hwpjpkT" width="595" height="485" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:1px solid #CCC; border-width:1px; margin-bottom:5px; max-width: 100%;" allowfullscreen></iframe>

                                    <figcaption><a href="//www.slideshare.net/GRNsight/varshneya-samdarshi-lmusymposium2016" title="Varshneya samdarshi lmu_symposium_2016" target="_blank">Poster</a> from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>
<br><br>
                                <p class="abstract">
                                    <b><a href="http://academics.lmu.edu/media/lmuacademics/undergraduateresearchsymposium/documents/ei8ht_Symposium_Long_Program_16_Secured.pdf">Loyola Marymount University 8<sup>th</sup> Annual Undergraduate Research Symposium</a></b><br><br>
                                    March 19, 2016<br>
                                    Test-driven development improves GRNsight: a web application for visualizing models of gene regulatory networks<br>
                                    <b>Anindita Varshneya, Mihir Samdarshi</b>, Kam D. Dahlquist, John David N. Dionisio, and Ben G. Fitzpatrick<br><br>

                                    Abstract: GRNsight is an open source web application and service for visualizing models of gene regulatory networks (GRNs). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN, can mathematically model the dynamics of GRNs. GRNsight automatically lays out the network graph based on GRNmap output spreadsheets. GRNsight uses pointed and blunt arrowheads, and colors the edges and adjusts their thicknesses based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. Visualizations can be modified through manual node dragging and sliders that adjust the force graph parameters. We have now implemented an exhaustive unit testing framework using Mocha and the Chai assertion library to perform test-driven development where unit tests are written before new functionality is coded. This framework consists of over 160 automated unit tests that examine over 450 test files to ensure that the program is running as expected. Error and warning messages inform the user what happened, the source of the problem, and possible solutions. The completion of the testing framework marks the close of development for version 1 (the current release stands at version 1.12). In version 2.0 of GRNsight, a new feature will be implemented that colors the nodes (genes) based on expression data provided by GRNmap. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.
                                </p>

                                <figure class="publication">
                                    <iframe src="//www.slideshare.net/slideshow/embed_code/key/neLFcr6hwpjpkT" width="595" height="485" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:1px solid #CCC; border-width:1px; margin-bottom:5px; max-width: 100%;" allowfullscreen></iframe>

                                    <figcaption><a href="//www.slideshare.net/GRNsight/varshneya-samdarshi-lmusymposium2016" title="Varshneya samdarshi lmu_symposium_2016" target="_blank">Poster</a> from <a href="//www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>

                                <p class="abstract">
                                    <b>40<sup>th</sup> Annual West Coast Biological Sciences Undergraduate Research Conference</b><br><br>
                                    April 25, 2015 <br>
                                    Improvements to GRNsight: a Web Application for Visualizing Models of Gene Regulatory Networks<br>
                                    <b>Nicole Anguiano, Anindita Varshneya</b>, Kam D. Dahlquist, John David N. Dionisio, and Ben G. Fitzpatrick<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing models of gene regulatory networks (GRN). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN, can mathematically model GRNs. The general graph visualization tools that exist for displaying GRNs did not fulfill our needs, so we created GRNsight. GRNsight automatically lays out the network graph based on spreadsheets uploaded from GRNmap. GRNsight colors the edges and adjusts their thicknesses based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. GRNsight uses pointed and blunt arrowheads to indicate activation and repression, respectively. Visualizations can be modified through manual node dragging and sliders that adjust the force layout parameters. Since version 1.0, GRNsight has acquired significant user interface improvements, bug fixes, extended robustness to errors, and aesthetic refinements. Demo spreadsheets are now available for users who do not have their own models. GRNsight's current release is version 1.10, with beta version 1.11 available for testing new features. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.
                                </p>

                                <figure class="publication">
                                    <iframe src="//www.slideshare.net/slideshow/embed_code/46240444" width="476" height="400" frameborder="0" marginwidth="0" marginheight="0" scrolling="no"></iframe>

                                    <figcaption><a href="http://www.slideshare.net/GRNsight/anguiano-varshneya-lmusymposium2015" title="Anguiano varshneya lmu_symposium_2015" target="_blank">Poster</a> from <a href="http://www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>

                                <p class="abstract">
                                    <b>Loyola Marymount University 7<sup>th</sup> Annual Undergraduate Research Symposium</b><br><br>
                                    March 21, 2015<br>
                                    Improvements to GRNsight: a Web Application for Visualizing Models of Gene Regulatory Networks<br>
                                    <b>Nicole Anguiano, Anindita Varshneya</b>, Kam D. Dahlquist, John David N. Dionisio, and Ben G. Fitzpatrick<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing models of gene regulatory networks (GRN). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNs can be mathematically modeled by GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN. The general graph visualization tools that exist for displaying GRNs did not fulfill our needs, so we created GRNsight. GRNsight automatically lays out the network graph based on spreadsheets uploaded from GRNmap. It is written in JavaScript, with diagrams facilitated by D3.js. Node.js and the Express framework handle server-side functions. GRNsight’s diagrams are based on D3.js’s force graph layout algorithm, which was then extensively customized to support the specific needs of GRN visualization. GRNsight colors the edges and adjusts their thicknesses based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. GRNsight uses pointed and blunt arrowheads to indicate activation and repression, respectively. Visualizations can be modified through manual node dragging and sliders that adjust D3.js’s force layout parameters. Since version 1.0, GRNsight has acquired significant user interface improvements, bug fixes, extended robustness to errors in file formats, and aesthetic refinements, particularly for edge markers. Demo spreadsheets are now available for users who do not have their own models. GRNsight's current release is version 1.8, with beta version 1.9 available for testing new features. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.
                                </p>

                                <figure class="publication">
                                    <iframe src="//www.slideshare.net/slideshow/embed_code/46240444" width="476" height="400" frameborder="0" marginwidth="0" marginheight="0" scrolling="no"></iframe>

                                    <figcaption><a href="http://www.slideshare.net/GRNsight/anguiano-varshneya-lmusymposium2015" title="Anguiano varshneya lmu_symposium_2015" target="_blank">Poster</a> from <a href="http://www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</div>
                                </figure>

                                <p class="abstract">
                                    <b>5<sup>th</sup> Annual Southern California Systems Biology Conference</b><br><br>
                                    January 31, 2015<br>
                                    Improvements to GRNsight: a Web Application for Visualizing Models of Gene Regulatory Networks<br>
                                    <b>Nicole Anguiano</b>, Anindita Varshneya, Britain J. Southwick, Kam D. Dahlquist, John David N. Dionisio, and Ben G. Fitzpatrick<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing models of gene regulatory networks (GRN). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNs can be mathematically modeled by applications such as GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN. GRNsight automatically lays out the network graph based on spreadsheets uploaded from GRNmap. It is written in JavaScript, with diagrams facilitated by D3.js. Node.js and the Express framework handle server-side functions. GRNsight’s diagrams are based on D3.js’s force graph layout algorithm, which was then extensively customized to support the specific needs of GRN visualization. GRNsight colors the edges and adjusts their thicknesses based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. GRNsight uses pointed and blunt arrowheads to indicate activation and repression, respectively. When a user mouses over an edge, the numerical value of the weight parameter is displayed. Visualizations can be modified through manual node dragging and sliders that adjust D3.js’s force layout parameters. Since version 1.0, GRNsight has acquired significant user interface improvements, bug fixes, extended robustness to errors in file formats, and aesthetic refinements, particularly for edge markers. Demo spreadsheets are now available for users who do not have their own models. GRNsight's current release is version 1.8, with beta version 1.9 available for testing new features. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.
                                </p>

                                <figure class="publication">
                                    <iframe src="//www.slideshare.net/slideshow/embed_code/42279842" width="476" height="400" frameborder="0" marginwidth="0" marginheight="0" scrolling="no"></iframe>

                                    <figcaption><a href="http://www.slideshare.net/GRNsight/anguiano-varshneya-sccurposter20141122" title="Anguiano varshneya sccur-poster_20141122" target="_blank">Poster</a> from <a href="http://www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>

                                <p class="abstract">
                                    <b>2014 Southern California Conference for Undergraduate Research</b><br><br>
                                    November 22, 2014<br>
                                    GRNsight: A Web Application for Visualizing Models of Gene Regulatory Networks<br>
                                    <b>Nicole Anguiano, Anindita Varshneya</b>, John David N. Dionisio, Kam D. Dahlquist, Ben G. Fitzpatrick<br><br>

                                    Abstract: GRNsight is a web application and service for visualizing models of gene regulatory networks (GRN). A gene regulatory network consists of genes, transcription factors, and the regulatory connections between them which govern the level of expression of mRNA and protein from genes. GRNs can be mathematically modeled by applications such as GRNmap, a MATLAB program that performs parameter estimation and forward simulation of a differential equations model of a GRN. The general graph visualization tools that exist for displaying GRNs did not fulfill our needs, so we created GRNsight. GRNsight automatically lays out the network graph based on spreadsheets uploaded from GRNmap. It is written in JavaScript, with diagrams facilitated by D3.js, a data visualization library. Node.js and the Express framework handle server-side functions. GRNsight’s diagrams are based on D3.js’s force graph layout algorithm, which was then extensively customized to support the specific needs of GRN visualization. GRNsight colors the edges and adjusts their thickness based on the sign (activation or repression) and magnitude of the GRNmap weight parameter. GRNsight uses arrowheads and perpendicular line segments to indicate activation and repression, respectively. When a user mouses over an edge, the numerical value of the weight parameter is displayed. Visualizations can be modified through manual node dragging and sliders that adjust D3.js’s force layout parameters. GRNsight follows an open development model with rigorous documentation of requirements and issues on GitHub. Since version one, GRNsight has acquired significant user interface improvements, bug fixes, extended robustness to errors in file formats, and aesthetic refinements, particularly for edge markers. Demo spreadsheets are now available for users who do not have their own models. GRNsight's current release is version 1.6, with beta version 1.7 available for testing new features. GRNsight is available at <a href="http://dondi.github.io/GRNsight/">http://dondi.github.io/GRNsight/</a>.
                                </p>

                                <figure class="publication">
                                    <iframe src="//www.slideshare.net/slideshow/embed_code/42279842" width="476" height="400" frameborder="0" marginwidth="0" marginheight="0" scrolling="no"></iframe>

                                    <figcaption><a href="http://www.slideshare.net/GRNsight/anguiano-varshneya-sccurposter20141122" title="Anguiano varshneya sccur-poster_20141122" target="_blank">Poster</a> from <a href="http://www.slideshare.net/GRNsight" target="_blank">GRNsight</a> on SlideShare</figcaption>
                                </figure>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
