# GRNsight

[![DOI](https://zenodo.org/badge/16195791.svg)](https://zenodo.org/badge/latestdoi/16195791)
[![Node.js CI](https://github.com/dondi/GRNsight/actions/workflows/node.js.yml/badge.svg)](https://github.com/dondi/GRNsight/actions/workflows/node.js.yml)
[![Coverage Status](https://coveralls.io/repos/github/dondi/GRNsight/badge.svg?branch=beta)](https://coveralls.io/github/dondi/GRNsight?branch=beta)

http://dondi.github.io/GRNsight/

GRNsight is an open source web application and service for visualizing models of small-scale gene regulatory networks (GRNs) and protein-protein interaction networks (PPIs). GRNsight is a joint project of the [Loyola Marymount University](http://www.lmu.edu) Bioinformatics and Biomathematics Groups, headed by [Dr. Kam Dahlquist](http://myweb.lmu.edu/kdahlqui/index.htm), [Dr. John David N. Dionisio](http://myweb.lmu.edu/dondi/), and [Dr. Ben G. Fitzpatrick](http://myweb.lmu.edu/bfitzpatrick/). Undergraduate students initiated the development of GRNsight in Spring 2014, including Britain Southwick (Computer Science, ’14) and Nicki Anguiano (Computer Science, ’16), with consultation from Katrina Sherbina (Biomathematics, ’14). GRNsight has been in continuous development since that time by successive generations of undergraduates. For a list of other former and current developers, please see our [People](http://dondi.github.io/GRNsight/people.html) page.

A gene regulatory network (GRN) consists of genes, transcription factors, and the regulatory connections between them, which govern the level of expression of mRNA and protein from those genes. GRNs can be mathematically modeled and simulated by applications such as our sister project [GRNmap](http://kdahlquist.github.io/GRNmap/), a MATLAB program that estimates the parameters and performs forward simulations of a differential equations model of a GRN. Computer representations of GRNs, such as the models output by GRNmap, are in the form of a tabular spreadsheet (adjacency matrix) that is not easily interpretable. Ideally, GRNs should be displayed as diagrams (graphs) detailing the regulatory relationships (edges) between each gene (node) in the network. To address this need, we developed GRNsight.

GRNsight allows users to upload Excel workbooks generated by GRNmap or Excel, SIF, or GraphML files created by the user according to our [Documentation]([http://kdahlquist.github.io/GRNmap/](https://dondi.github.io/GRNsight/documentation.html)). GRNsight automatically creates and displays a graph of the GRN model through a physics-based force simulation or in a grid layout. The application colors the edges, adjusts their thickness, and applies arrowhead or blunt-end markers based on the sign (activation or repression) and the strength (magnitude) of the regulatory relationship, respectively. Users without their own network data can load a GRN by selecting genes from the backend database that contains regulation data from the <i>Saccharomyces</i> Genome Database (SGD). Nodes can be colored based on time course gene expression data supplied in the input workbook or from select public datasets stored in the backend database. Finally, GRNsight allows the user to modify the graph display by panning, zooming, and moving nodes to define the best visual layout for the network. GRNsight's functionality has been extended for displaying PPI networks with undirected edges, either uploaded by the user or loaded from our backend database which contains protein-protein physical interaction data from SGD. 

Most of GRNsight is written in JavaScript. HTTP requests are handled using Node.js and the Express framework. Graphs are generated through D3.js, a JavaScript data visualization library.

While the GRNsight database only contains budding yeast data at this time, GRNsight can display user-uploaded data from any species. And although originally designed for gene regulatory networks, we believe that GRNsight has general applicability for displaying any small, unweighted or weighted network with directed or undirected edges for systems biology or other application domains. 

Most users will want to access GRNsight through the web application at http://dondi.github.io/GRNsight/. The source code is available for developers who wish to run their own instance of the GRNsight web service and/or web client.

Documentation on how to use GRNsight is found at https://dondi.github.io/GRNsight/documentation.html, with additional information on the wiki here: https://github.com/dondi/GRNsight/wiki.

If you use GRNsight in your work, please cite:

Dahlquist, K.D., Dionisio, J.D.N., Fitzpatrick, B.G., Anguiano, N.A., Varshneya, A., Southwick, B.J., Samdarshi, M. (2016) GRNsight: a web application and service for visualizing models of small- to medium-scale gene regulatory networks. _PeerJ Computer Science_ 2:e85. [DOI: 10.7717/peerj-cs.85](https://doi.org/10.7717/peerj-cs.85).
